pkg/source/resource.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package source
+
+import (
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest"
+)
+
+func FetchResourceObjects(builder *resource.Builder, namespace string, labelSelectorParam string, types []string) (runtime.Object, error) {
+	r := builder.Unstructured().
+		NamespaceParam(namespace).
+		LabelSelectorParam(labelSelectorParam).
+		ResourceTypeOrNameArgs(true, types...).
+		Latest().
+		Flatten().
+		TransformRequests(func(req *rest.Request) {
+			req.SetHeader("Accept", strings.Join([]string{
+				fmt.Sprintf(runtime.ContentTypeJSON+";as=Table;v=%s;g=%s", metav1.SchemeGroupVersion.Version, metav1.GroupName),
+				runtime.ContentTypeJSON,
+			}, ","))
+		}).
+		Do()
+	infos, err := r.Infos()
+	if err != nil {
+		return nil, fmt.Errorf("%s", err)
+	}
+	return decodeIntoTable(infos)
+}
+
+func decodeIntoTable(info []*resource.Info) (runtime.Object, error) {
```

Comment:
goal of the func is to decode an Info resource  into a runtime object so instead of passing an array, just `func decodeIntoObject(info *resource.Info) (runtime.Object, error) {`

caller can decide to pass only 1 info object in this case. 

```
if len(infos) >= 1 { or // for i:= 0; i<1;i++ {
  obj, err := decodeIntoTable(infos[0])
}
return nil, nil

Reply:
Done